nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/TailFile.java

```diff
@@ -1172,19 +1188,28 @@ private boolean recoverRolledFiles(final ProcessContext context, final ProcessSe
                 final File firstFile = rolledOffFiles.get(0);
 
                 final long startNanos = System.nanoTime();
+                final Boolean reReadOnNul = context.getProperty(REREAD_ON_NUL).asBoolean();
                 if (position > 0) {
-                    try (final InputStream fis = new FileInputStream(firstFile);
-                            final CheckedInputStream in = new CheckedInputStream(fis, new CRC32())) {
-                        StreamUtils.copy(in, new NullOutputStream(), position);
+                    try (final FileInputStream fis = new FileInputStream(firstFile)) {
```

Comment:
I think we don't need `readLines` here because we already processed to content up to this point.

Reply:
Thanks @tpalfy , I have updated the change according to your comment. I also extracted a couple of repeating lines from the `readLines()` method.